src/daterangepicker/daterangepicker.component.ts

```diff
@@ -1,1230 +1,1350 @@
+  /** Flag to display apply button */
+  @Input() showCancelButton = false;
+  /** Flag to display apply button */
+  @Input() showApplyButton = false;
+  /** Flag to display clear button */
+  @Input() showClearButton = false;
+
+  // CALENDAR BEHAVIOR
+  /** Flag to keep the calendar open after choosing a range */
+  @Input() keepCalendarOpeningWithRange = false;
+  /** Flag to display the range label on input */
+  @Input() showRangeLabelOnInput = false;
+  /** Flag to allow selection range from end date first */
+  @Input() customRangeDirection = false;
+  /** Flag to lock start date and change only the end date */
+  @Input() lockStartDate = false;
+  /** Flag to update input when selecting a date/range  */
+  @Input() autoUpdateInput = true;
+  /** Flag to display the ranges with the calendar*/
+  @Input() alwaysShowCalendars = false;
+  /** Flag to link both calendars */
+  @Input() linkedCalendars = false;
+  /** Close datepicker when auto apply */
+  @Input() closeOnAutoApply = true;
+  /** Flag to auto apply changes on select */
+  @Input() autoApplyChanges = false;
+
+  @Input() maxSpan = false;
+
+  // timepicker variables
+  @Input() timePicker: Boolean = false;
+  @Input() timePicker24Hour: Boolean = false;
+  @Input() timePickerIncrement = 1;
+  @Input() timePickerSeconds: Boolean = false;
+  // end of timepicker variables
+
+  /** Set calendar locale settings */
+  private _locale: LocaleConfig = {};
+  @Input() set locale(value) {
+    this._locale = {...this._localeService.config, ...value};
+  }
+
+  get locale(): any {
+    return this._locale;
+  }
+
+  /** Set custom ranges */
+  private _ranges: any = {};
+  @Input() set ranges(value) {
+      this._ranges = value;
+      this.renderRanges();
+  }
+
+  get ranges(): any {
+      return this._ranges;
+  }
+
+  // CUSTOM CSS
+  @Input() firstMonthDayClass: string = null;
+  @Input() lastMonthDayClass: string = null;
+  @Input() emptyWeekRowClass: string = null;
+  @Input() firstDayOfNextMonthClass: string = null;
+  @Input() lastDayOfPreviousMonthClass: string = null;
+
+  chosenRange: string;
+  rangesArray: Array<any> = [];
+
+  /** Calendar state */
+  isShown = false;
+  inline = true;
+  leftCalendar: any = {};
+  rightCalendar: any = {};
+  showCalendarInRanges = false;
+
+  options: any = {}; // should get some opt from user
+
+  /** Event on date selected */
+  @Output() choosedDate: EventEmitter<Object>;
+  /** Event on range clicked */
+  @Output() rangeClicked: EventEmitter<Object>;
+  /** Event on dates updated */
+  @Output() datesUpdated: EventEmitter<Object>;
+  /** Event on start date changed */
+  @Output() startDateChanged: EventEmitter<Object>;
+  /** Event on end date changed */
+  @Output() endDateChanged: EventEmitter<Object>;
+
+  @ViewChild('pickerContainer') pickerContainer: ElementRef;
+
+  @HostListener('click', ['$event'])
+  stopClick(event: Event) {
+    event.preventDefault();
+    event.stopPropagation();
+  }
```

Comment:
this is used? you are using the directive in the html.. do you need this one too?